base/broadcast.jl

```diff
@@ -85,12 +113,13 @@ end
 # `newindex` above) for a particular array `A`, given the
 # broadcast_shape `shape`
 # `keep` is equivalent to map(==, indices(A), shape) (but see #17126)
-newindexer(shape, x::Number) = (), ()
-@inline newindexer(shape, A) = newindexer(shape, indices(A))
-@inline newindexer(shape, indsA::Tuple{}) = (), ()
-@inline function newindexer(shape, indsA::Tuple)
+newindexer(shape, x) = (), ()
+newindexer(shape, t::Tuple) = _newindexer(shape, (OneTo(length(t)),))
+@inline newindexer(shape, A::AbstractArray) = _newindexer(shape, indices(A))
```

Comment:
Aren't these three cases equivalent to `newindexer(shape, x) = (), broadcast_shape(x)`?